src/amp-story-player/amp-story-entry-point/amp-story-entry-point-impl.js

```diff
@@ -84,6 +113,170 @@ export class AmpStoryEntryPoint {
+    cardsContainer.classList.add('entry-points');
+    this.rootEl_.append(cardsContainer);
+
+    for (let i = 0; i < this.stories_.length; i++) {
+      const story = this.stories_[i];
+      const src = story.getAttribute('data-poster-portrait-src');
+
+      const cardContainer = this.doc_.createElement('div');
+      cardContainer.classList.add('entry-point-card-container');
+      cardsContainer.append(cardContainer);
+
+      if (styleClass !== 'circular-entry-point') {
+        this.initializeCardTitle_(cardContainer, story);
+        this.initializeCardLogo_(cardContainer);
+      }
+
+      this.initializeCard_(cardContainer, src, styleClass, story.href, i);
+    }
+  }
+
+  /**
+   * Initializes the card's title to be displayed.
+   * @param {Element} cardContainer
+   * @param {HTMLAnchorElement} story
+   */
+  initializeCardTitle_(cardContainer, story) {
+    const title = this.doc_.createElement('span');
+    title.classList.add('entry-point-card-headline');
+    title.textContent = story.querySelector('span').textContent;
+    cardContainer.append(title);
+  }
+
+  /**
+   * Initializes the logo to be displayed.
+   * @param {Element} cardContainer
+   */
+  initializeCardLogo_(cardContainer) {
+    const src = this.element_.getAttribute('logo');
+    if (!src) {
+      return;
+    }
+
+    const logo = this.doc_.createElement('img');
+    logo.classList.add('entry-point-card-logo');
+    logo.src = src;
+    cardContainer.append(logo);
+  }
+
+  /**
+   * Initializes an image element to be displayed.
+   * @param {Element} cardContainer
+   * @param {string} src
+   * @param {string} styleClass
+   * @param {string} href
+   * @param {number} index
+   * @private
+   */
+  initializeCard_(cardContainer, src, styleClass, href, index) {
+    const card = this.doc_.createElement('img');
+    card.src = src;
+    card.classList.add(styleClass);
+    card.classList.add('entry-point-card-img');
+    this.setCardWidthHeight_(card, styleClass);
+    this.onCardClick_(card, href, index);
+    cardContainer.append(card);
+  }
+
+  /**
+   * Sets the correct height and width for poster based on style.
+   * @param {HTMLImageElement} card
+   * @param {string} styleClass
+   * @private
+   */
+  setCardWidthHeight_(card, styleClass) {
+    card.height = 233;
+    if (styleClass === 'circular-entry-point') {
+      card.width = 100;
+      card.height = 100;
+    }
+  }
+
+  /**
+   * Reacts to on click event for an entry point.
+   * @param {HTMLImageElement} card
+   * @param {string} href
+   * @param {number} index
+   * @private
+   */
+  onCardClick_(card, href, index) {
+    card.addEventListener('click', () => {
+      this.player_.show(href);
+      this.element_.dispatchEvent(
+        createCustomEvent(this.win_, 'entryPointClicked', {href, index})
+      );
+    });
+  }
+
+  /**
+   * Creates arrows to be displayed.
+   * @private
+   * */
+  initializeCarouselArrow_() {
```

Comment:
Carousel arrows are not needed on mobile, so there should be a check that if on mobile, also return early.